Quantum-Control-Applications/Superconducting/Multiple Flux Tunable Transmons/Use Case 2 - Two Qubit Randomized Benchmarking/README.md

```diff
@@ -0,0 +1,238 @@
+
+```python
+def meas():
+    threshold0 = 0.3 #example value
+    threshold1 = 0.3 #example value
+    rr0_name = f"qubit{q0}_rr"
+    rr1_name = f"qubit{q1}_rr"
+    Iq0 = declare(fixed)
+    Qq0 = declare(fixed)
+    Iq1 = declare(fixed)
+    Qq1 = declare(fixed)
+    state0 = declare(bool)
+    state1 = declare(bool)
+    measure("readout", rr0_name, None,
+            dual_demod.full("w1", "out1", "w2", "out2", Iq0),
+            dual_demod.full("w3", "out1", "w1", "out2", Qq0)
+            )
+    measure("readout", rr1_name, None,
+            dual_demod.full("w1", "out1", "w2", "out2", Iq1),
+            dual_demod.full("w3", "out1", "w1", "out2", Qq1)
+            )
+    assign(state0, Iq0 > threshold0)
+    assign(state1, Iq1 > threshold1)
+    return state0, state1
+```
+
+## Execution and Results
+Using the *TwoQubitRb* class we can construct the experiment by specifying the previously defined single- and two-qubit gate functions, as well as the preparation and measurement protocols. The class translates the native gate set to Clifford operations using [*Google Cirq*](https://quantumai.google/cirq), generates the gate sequences and finds the inverse that resets the qubits to the state |00>. When gate errors occur, the inverse gate cannot reset the unitary circuit to the ground state and we will see a fidelity decrease, typically for increasing circuit depth. Note, that it is possible to provide the √iSWAP or CNOT gate by adding them to the two-qubit dictionary to allow an optimized gate decomposition into Clifford gates. If other native gates are implemented, the decomposition has to be added to [*gates.py*](https://github.com/qua-platform/qua-libs/blob/2qb-RB-usecase/Quantum-Control-Applications/Superconducting/Multiple%20Flux%20Tunable%20Transmons/Use%20Case%202%20-%20Two%20Qubit%20Randomized%20Benchmarking/two_qubit_rb/gates.py#L43).
+
+```python
+rb = TwoQubitRb(config, bake_phased_xz, {"CZ": bake_cz}, prep, meas, verify_generation=True)
+```
+Before running the experiment, we have to specify the utilized OPX-cluster by creating the *qmm* object with the *QuantumMachinesManager* class. Then, the experiment is executed by calling the run method of the previously generated two-qubit RB program *rb*. Here, we also add important benchmarking parameters like circuit depth (*circuit_depths*), how many different circuits we would like to run per depth (*num_circuits_per_depth*) and how often we we would like to run every circuit (*num_shots_per_circuit*).
+
+```python
+qmm = QuantumMachinesManager('127.0.0.1',8080)
+res = rb.run(qmm, circuit_depths=[1, 2, 3, 4, 5], num_circuits_per_depth=50, num_shots_per_circuit=1000)
```

Comment:
Are there any limitations in terms of depths (can I put 1000 depths up to 10_000 for instance)?
Same for the other parameters

Reply:
I don't think so since this uses the input stream, but I'll verify